pMim - Integrating RNA-Seq, miRNA-Seq and pathway information
========================================================

```{r include=FALSE}
opts_chunk$set(message=FALSE, warning=FALSE)
```


The following provides an example of how to use pMim. The markdown code that generated this html file can be doun on my github account at <a href="https://github.com/ellispatrick/sydSeq/blob/master/Examples/pMim/pMimExample.Rmd" target="_blank"> pMimExample.Rmd </a>. Start by reading in the RNA-Seq and miRNA-Seq data.


```{r}
### Load packages
library(sydSeq) #For pMimCor
library(multiMiR) #For getting miRNA target genes easily
library(goseq) #For getting KEGG pathways easily
library(KEGG.db)

### Load Notch2 Knockout counts for use as an example.

load('counts.RData')
#This loaded the gene counts "counts" and the miRNA counts "countsMi".

### For simplicity, while obviously not ideal, for the following we will work with TMM normalized data. We will also restrict to genes with average count over 20.

tmm = function(DAT){
  TMM = DAT[1,]
  for(i in (1:dim(DAT)[2])){
    Data = cbind(DAT[,i],rowMeans(DAT))
    X = Data[,]
    X[rowSums(Data<100)>0,] = NA
    Y = log(X[,1])-log(X[,2])
    TMM1 = mean(Y,na.rm = TRUE,trim = .3)
    TMM[i] = exp(TMM1)
  }
  TMM = ((TMM)/exp(mean(log(TMM))))
  TMM = 1/TMM
  TMM
  }

Data = counts[rowSums(counts)>=20,]
Data = Data*matrix(tmm(Data),dim(Data)[1],dim(Data)[2],byrow = TRUE)

DataMi = countsMi[rowSums(countsMi)>=20,]
DataMi = DataMi*matrix(tmm(DataMi),dim(DataMi)[1],dim(DataMi)[2],byrow = TRUE)

```


The pathway and miRNA target information can then be read in. Here we use KEGG for the pathway information and TargetScan for the miRNA target predictions.


```{r}
# Read in KEGG pathways and Target matrix.

###KEGG can be loaded as follows
kegg = getgo(rownames(Data), fetch.cats = "KEGG", genome = "mm10", id = "ensGene")
kegg2ens = Biobase::reverseSplit(kegg)
kname = mget(names(kegg2ens), env = KEGGPATHID2NAME)
names(kegg2ens) = kname
#kegg2ens is our list of pathways


### miRNA Targets can be loaded as follows

x = get.multimir(org = "mmu", mirna = NULL, target = rownames(Data), disease.drug = NULL, 
    table = "targetscan", predicted.cutoff = 30, predicted.cutoff.type = "p", 
    summary = FALSE)
x = x[[1]]

# We do not have mature miRNA counts, so do the following.
miR = rownames(DataMi)
miR = sub("mir", "miR", miR)
miR = c(miR, paste(miR, "-3p", sep = ""), paste(miR, "-5p", sep = ""))
miR = split(miR, c(rownames(DataMi), rownames(DataMi), rownames(DataMi)))

targets = lapply(miR, function(z) unique(x[x[, "mature_mirna_id"] %in% z, "target_ensembl"]))
#targets is our list of miRNA targets

```

We are now in a position to run pMim and view the top ten mir-pathways that are identified.

```{r}
classes = substring(colnames(Data),1,2)
names(classes) = colnames(Data)

output = pMim(DataMi = DataMi, DataG = Data, classes = classes, targets = targets, pathways = kegg2ens)

head(output$Results,10)

```

We can also output the results into table which can be embedded in html code or their own html files.
An example of the output is give in <a href="pMimResults.html" target="_blank"> pMimResults.html </a>.

Clicking on the "genes" links will take you to the genes that lie in the selected mir-pathway. These tables generated by the googleVis package and can be sorted by clicking on the column headers. 


```{r results='asis'}
load('GeneSymbol.RData')

dir = 'mirpathways'
cutoff = 0.05
filename = 'pMimResults.html'
GeneSymbol = Symbol

htmlOutput = pMimHTML(output,dir = 'mirpathways',cutoff = 0.05,filename = 'pMimResults.html',GeneSymbol = Symbol,outputHTML=TRUE)
print(htmlOutput,type = 'html') 



```

